### ISSUE TITLE
Enhancing UX: Accessibility & Wallet Dropdown

### ISSUE THREAD

**Issue by annavilnid** [compolabs/spark-frontend#389]
<img width="216" alt="Image" src="https://github.com/user-attachments/assets/410d56fb-3ab2-4f0e-96d0-1e2432a9f6a2" />

1️⃣ The "Disconnect" button does not work with Fuelet Wallet.
2️⃣ When connecting a wallet, an incorrect balance is often displayed in the first few seconds (either zero or the balance of the previous wallet). It might be better to show a visual loading indicator or a skeleton until the actual data is available.
3️⃣ The balance appears to be an interactive element but is not.
4️⃣ Keyboard navigation is missing – dropdown elements and the trigger are not focusable via the Tab key because buttons and links are not implemented with the correct tags.
5️⃣ Icons are not labeled for screen readers – aria-label attributes are missing.
